haskell/linux/OS.hs

```diff
@@ -159,3 +180,107 @@ data Axis
 
 keyUnknown :: Key
 keyUnknown = KeyUnknown
+
+charToKey :: Char -> Maybe Key
+charToKey = \case
+    '0' -> Just Key0
+    '1' -> Just Key1
+    '2' -> Just Key2
+    '3' -> Just Key3
+    '4' -> Just Key4
+    '5' -> Just Key5
+    '6' -> Just Key6
+    '7' -> Just Key7
+    '8' -> Just Key8
+    '9' -> Just Key9
+    'a' -> Just KeyA
+    'b' -> Just KeyB
+    'c' -> Just KeyC
+    'd' -> Just KeyD
+    'e' -> Just KeyE
+    'f' -> Just KeyF
+    'g' -> Just KeyG
+    'h' -> Just KeyH
+    'i' -> Just KeyI
+    'j' -> Just KeyJ
+    'k' -> Just KeyK
+    'l' -> Just KeyL
+    'm' -> Just KeyM
+    'n' -> Just KeyN
+    'o' -> Just KeyO
+    'p' -> Just KeyP
+    'q' -> Just KeyQ
+    'r' -> Just KeyR
+    's' -> Just KeyS
+    't' -> Just KeyT
+    'u' -> Just KeyU
+    'v' -> Just KeyV
+    'w' -> Just KeyW
+    'x' -> Just KeyX
+    'y' -> Just KeyY
+    'z' -> Just KeyZ
+    '\'' -> Just KeyApostrophe
+    '\\' -> Just KeyBackslash
+    ',' -> Just KeyComma
+    '.' -> Just KeyDot
+    '=' -> Just KeyEqual
+    '`' -> Just KeyGrave
+    '{' -> Just KeyLeftbrace
+    '-' -> Just KeyMinus
+    '}' -> Just KeyRightbrace
+    ';' -> Just KeySemicolon
+    '/' -> Just KeySlash
+    ' ' -> Just KeySpace
+    _ -> Nothing
+
+keyToChar :: Key -> Maybe Char
+keyToChar = \case
```

Comment:
DRY this with the above. Maybe by constructing a top level list of pairs, and two maps.